api/errors.go

```diff
@@ -37,3 +40,105 @@ func NewResourceExhaustedError(msg string) error {
 func NewInternalError(msg string) error {
 	return NewGRPCError(codes.Internal, msg)
 }
+
+// ==================================================================
+// API CLIENT ERRORS
+// Note: These errors are currently used by api/clients.
+// Eventually it might be useful to use them across the entire codebase.
+// ==================================================================
+
+// Code below is adapted from https://github.com/aws/smithy-go/blob/main/errors.go
+
+// ErrorAPI is the most generic API and protocol agnostic error interface that we eventually
+// want every api clients returned errors to implement.
+//
+// This way, consumers of the api clients can tell what kind of error they are dealing with,
+// most broadly whether it is a client or server fault (see the ErrorFault type).
+type ErrorAPI interface {
+	error
+
+	// ErrorCode returns the error code for the API exception.
+	ErrorCode() ErrorCode
```

Comment:
Good idea: [b4b3441](https://github.com/Layr-Labs/eigenda/pull/828/commits/b4b34411c210de18b23bc9841a0b7186ce0856e3)

I guess my main concern was whether we eventually switch to grpc error codes instead. In that case we might want ErrorCode to be a string for eg instead of an int. But in the meantime agree using stdlib is best.

Reply:
Actually changed back in [8eeb3e0](https://github.com/Layr-Labs/eigenda/pull/828/commits/8eeb3e0968b90128578dfd494b469c7b56afdb81) to grpc error codes only. I think this whole tanged with ErrorAPI was a mistake... mea culpa.